lib/common/containers/ActiveUserRetriever.js

```diff
@@ -0,0 +1,41 @@
+// @flow
+import { useAuth0 } from '@auth0/auth0-react'
+// $FlowFixMe useEffect not recognized by flow.
+import { useEffect } from 'react'
+import { connect } from 'react-redux'
+
+import * as userActions from '../../manager/actions/user'
+
+/**
+ * Retrieves the active, authenticated user from auth0 and
+ * updates the redux state accordingly.
+ */
+const ActiveUserRetriever = ({ logout, receiveTokenAndProfile }) => {
+  const auth0 = useAuth0()
+  const { isAuthenticated, isLoading, user: profile } = auth0
+
+  // Any time the login state reported by auth0 changes,
+  // update the user info in the redux state.
+  useEffect(async () => {
+    if (isAuthenticated) {
+      const tokenResponse = await auth0.getAccessTokenSilently({
+        detailedResponse: true
+      })
+      // TRICKY: What we are passing to the backend is the id_token portion of the response.
+      // Unlike the access_token portion, the id_token contains the app_metadata info for the user.
+      receiveTokenAndProfile({ profile, token: tokenResponse.id_token })
+    } else if (!isLoading) {
+      logout(auth0)
+    }
+  }, [isAuthenticated, isLoading, profile])
```

Comment:
This component doesn't seem to be working to grab the user for me. Maybe that's just because I'm not fully getting signed in though. Did it work for you @miles-grant-ibigroup?

Reply:
It took a lot of fiddling with auth0 settings to get right. I think some sort of migration guide/plan is crucial to getting this PR through fully

Reply:
Actually, what appears to be a migration issue comes from the need to copy datatools permissions for the test client app in Auth0 I created for this PR. When this is merged, the idea is to switch the existing Auth0 app to use the new RS256 algorithm, in which case no user migration will be needed.

Reply:
auth0-react returns an error message if for some reason it is not able to process the login in the situation that Daniel describes above.  With https://github.com/ibi-group/datatools-ui/pull/834/commits/2e33a20b6f5e07577e55896a86202d7572d682f3, that message will be shown to help with troubleshooting.

Reply:
@daniel-heppner-ibigroup Try again and see if an error message is shown after logging in.